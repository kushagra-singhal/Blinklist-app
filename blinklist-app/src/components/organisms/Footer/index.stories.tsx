import React from "react";
import Footer from "./index";
import { ComponentStory, ComponentMeta } from '@storybook/react';

export default {
  title: "Organisms/Footer",
  component: Footer,
}as ComponentMeta<typeof Footer>;

const template: ComponentStory<typeof Footer> = (args:any) => <Footer {...args} />;
export const Footeromponent = template.bind({});
Footeromponent.args = {};