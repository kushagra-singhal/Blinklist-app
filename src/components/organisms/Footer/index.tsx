import { Container, Box } from "@mui/material";
import Logo from "../../molecules/Logo";
import Typography from "../../atoms/Typography";

const FooterComponent = (props: any) => {
  const data = [
    {
      heading: "Editorial",
      topics: [
        "Book lists",
        "What is Notfication?",
        "What to read next?",
        "Benefits of reading",
      ],
    },
    {
      heading: "Useful links",
      topics: [
        "Pricing",
        "Blinkist business",
        "Gift cards",
        "Blinkist magaine",
      ],
    },
    {
      heading: "Company",
      topics: ["About", "Carrer", "partners", "Code of Conduct"],
    },
  ];
  return (
    <Box
      sx={{
        height: "270px",
        width: "100%",
        backgroundColor: "#F1F6F4",
        display: "flex",
        position: "sticky",
        top: "100vh",
        padding: "24px 0px",
        marginTop: "24px",
      }}
    >
      <Container sx={{ display: "flex", justifyContent: "start" }}>
        <Box sx={{ marginRight: "150px" }}>
          <Logo
            height={32}
            width={32}
            url='/assets/logo.png'
            name='Blinkist'
            data-testid='logo'
          />
          <Typography
            variant='h6'
            sx={{ fontWeight: "bold", color: "#0365F2", fontSize: "24px" }}
            data-testid='slogan'
          >
            Big ideas in small packages
            <br />
            Start learning now
          </Typography>
        </Box>
        <Box sx={{ display: "flex", justifyContent: "end" }}>
          {data.map((curr) => {
            return (
              <Box
                sx={{
                  margin: "0px 32px",
                }}
                data-testid='topics'
              >
                <Typography
                  pb={2}
                  sx={{
                    fontWeight: "bold",
                    display: "flex",
                    justifyContent: "start",
                    color: "#03314B",
                    fontSize: "17px",
                  }}
                >
                  {curr.heading}
                </Typography>
                {curr.topics.map((currTopic) => {
                  return (
                    <Typography
                      pb={1}
                      sx={{
                        display: "flex",
                        justifyContent: "start",
                        color: "#6D787E",
                      }}
                    >
                      {currTopic}
                    </Typography>
                  );
                })}
              </Box>
            );
          })}
        </Box>
      </Container>
    </Box>
  );
};

export default FooterComponent;
