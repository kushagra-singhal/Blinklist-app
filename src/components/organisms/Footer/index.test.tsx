import Footer from '.';
import renderer from 'react-test-renderer';


test('renders Footer', () => {
    const component = renderer.create(
        <Footer />
    );
  
  expect(component).toMatchSnapshot();
});

