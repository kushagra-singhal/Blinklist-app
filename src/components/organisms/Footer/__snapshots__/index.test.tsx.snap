// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer 1`] = `
<div
  className="MuiBox-root css-1sp4l3i"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1imjkd-MuiContainer-root"
  >
    <div
      className="MuiBox-root css-315n33"
    >
      <div
        className="MuiBox-root css-70qvj9"
      >
        <img
          alt="not found"
          className="MuiBox-root css-xgff9b"
          src="/assets/logo.png"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 css-cl9qd7-MuiTypography-root"
        >
          Blinkist
        </p>
      </div>
      <h6
        className="MuiTypography-root MuiTypography-h6 css-1dzacpr-MuiTypography-root"
        data-testid="slogan"
      >
        Big ideas in small packages
        <br />
        Start learning now
      </h6>
    </div>
    <div
      className="MuiBox-root css-ckyhk3"
    >
      <div
        className="MuiBox-root css-6iaxro"
        data-testid="topics"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ohx2p-MuiTypography-root"
        >
          Editorial
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Book lists
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          What is Notfication?
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          What to read next?
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Benefits of reading
        </p>
      </div>
      <div
        className="MuiBox-root css-6iaxro"
        data-testid="topics"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ohx2p-MuiTypography-root"
        >
          Useful links
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Pricing
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Blinkist business
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Gift cards
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Blinkist magaine
        </p>
      </div>
      <div
        className="MuiBox-root css-6iaxro"
        data-testid="topics"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ohx2p-MuiTypography-root"
        >
          Company
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          About
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Carrer
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          partners
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Code of Conduct
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`render Footer 1`] = `
<div
  className="MuiBox-root css-1sp4l3i"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1imjkd-MuiContainer-root"
  >
    <div
      className="MuiBox-root css-315n33"
    >
      <div
        className="MuiBox-root css-70qvj9"
      >
        <img
          alt="not found"
          className="MuiBox-root css-xgff9b"
          src="/assets/logo.png"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 css-cl9qd7-MuiTypography-root"
        >
          Blinkist
        </p>
      </div>
      <h6
        className="MuiTypography-root MuiTypography-h6 css-1dzacpr-MuiTypography-root"
        data-testid="slogan"
      >
        Big ideas in small packages
        <br />
        Start learning now
      </h6>
    </div>
    <div
      className="MuiBox-root css-ckyhk3"
    >
      <div
        className="MuiBox-root css-6iaxro"
        data-testid="topics"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ohx2p-MuiTypography-root"
        >
          Editorial
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Book lists
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          What is Notfication?
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          What to read next?
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Benefits of reading
        </p>
      </div>
      <div
        className="MuiBox-root css-6iaxro"
        data-testid="topics"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ohx2p-MuiTypography-root"
        >
          Useful links
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Pricing
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Blinkist business
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Gift cards
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Blinkist magaine
        </p>
      </div>
      <div
        className="MuiBox-root css-6iaxro"
        data-testid="topics"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ohx2p-MuiTypography-root"
        >
          Company
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          About
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Carrer
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          partners
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Code of Conduct
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Footer 1`] = `
<div
  className="MuiBox-root css-1sp4l3i"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1imjkd-MuiContainer-root"
  >
    <div
      className="MuiBox-root css-315n33"
    >
      <div
        className="MuiBox-root css-70qvj9"
      >
        <img
          alt="not found"
          className="MuiBox-root css-xgff9b"
          src="/assets/logo.png"
        />
        <p
          className="MuiTypography-root MuiTypography-body1 css-cl9qd7-MuiTypography-root"
        >
          Blinkist
        </p>
      </div>
      <h6
        className="MuiTypography-root MuiTypography-h6 css-1dzacpr-MuiTypography-root"
        data-testid="slogan"
      >
        Big ideas in small packages
        <br />
        Start learning now
      </h6>
    </div>
    <div
      className="MuiBox-root css-ckyhk3"
    >
      <div
        className="MuiBox-root css-6iaxro"
        data-testid="topics"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ohx2p-MuiTypography-root"
        >
          Editorial
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Book lists
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          What is Notfication?
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          What to read next?
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Benefits of reading
        </p>
      </div>
      <div
        className="MuiBox-root css-6iaxro"
        data-testid="topics"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ohx2p-MuiTypography-root"
        >
          Useful links
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Pricing
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Blinkist business
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Gift cards
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Blinkist magaine
        </p>
      </div>
      <div
        className="MuiBox-root css-6iaxro"
        data-testid="topics"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-ohx2p-MuiTypography-root"
        >
          Company
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          About
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Carrer
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          partners
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 css-e3ldce-MuiTypography-root"
        >
          Code of Conduct
        </p>
      </div>
    </div>
  </div>
</div>
`;
